### PR TITLE
Syndicate Lavaland Rework

### PR DESCRIPTION
**Pull request by unit0016**
## About The Pull Request
Replicates (but not copies) oldbase's Syndicate Lavaland base, with xenobiology and hydroponics, golem equipment vendor, ORM and Silo boards, etc. This is a bit of a special case considering it's technically something we had on oldbase - yet also a ruin, _and_ something /tg/ won't accept. I've done my best to modularize it regardless of the documentation being non-existent. As a bonus, this comes with a ice moon variant. This PR also fixes the windows on the trade ship, because my consistency dammit!
https://cdn.discordapp.com/attachments/640760727489085450/790171436093538364/unknown.png

## Why It's Good For The Game
Ghostroles are a fun way for players that have been round removed or are otherwise bored of the on-station experience to continue playing and benefit the station and round as a whole, if played in the right hands. Plus I wanna bring back W-T-R.

## Changelog
:cl:
tweak: The Syndicate have kindly updated planetary communications outposts...
add: ...And also begun operations on the wastes of Lavaland's Ice Moon.
server: Due to the lavaland generation refactor, trees, bushes, ETC can and WILL spawn inside the icemoon variant. This is an upstream issue and effects all ice ruins.
/:cl:

### CODE REVIEW

**Comment by Arctaisia** [comment]
I can bring back Arvai properly now :eyes:

**Comment by unit0016** [comment]
Noticed a minor issue with how I set up the areas on the icemoon version. Going to fix quickly.

**Comment by Azarak** [comment]
Remove the magic mirror as soon we'll have support for pref characters as ghost roles

**Comment by unit0016** [comment]
Really? That's cool as hell! Removing the mirror.

**Comment by Azarak** [comment]
Can you move the ruin definition to the mapping module we've got in modular_skyrat? Aswell as putting the maps in there under a new /maps/ folder. You can change a 'prefix' variable of the ruin type to a different folder

**Comment by Azarak** [comment]
@KathrinBailey Mapped changes look fine? I personally think it's fine to give them xenobiology, as it potentially gives them a lot of fun stuff to do

**Comment by unit0016** [comment]
Done. Mimicked the main folder's hierarchy to maintain consistency while I was at it.

**Comment by unit0016** [comment]
just realized, it would be a lot cleaner if I just blacklisted the original in config and not the space ruin code itself. Whoops. I'll do that when I'm free.

**Comment by Azarak** [comment]
> 
> 
> just realized, it would be a lot cleaner if I just blacklisted the original in config and not the space ruin code itself. Whoops. I'll do that when I'm free.

No, it's better this way. Keep it

File: code/datums/ruins/lavaland.dm [hunk 1]
```diff
@@ -52,15 +52,15 @@
 	suffix = "lavaland_surface_ash_walker1.dmm"
 	cost = 20
 	allow_duplicates = FALSE
-
+/* Skyrat Edit Removal Begin - Mapping
```
Comment (Azarak) [inline code/datums/ruins/lavaland.dm]:
```suggestion
//SKYRAT EDIT REMOVAL BEGIN - MAPPING
/*
```

File: code/datums/ruins/lavaland.dm [hunk 2]
```diff
@@ -52,15 +52,15 @@
...
 /datum/map_template/ruin/lavaland/syndicate_base
 	name = "Syndicate Lava Base"
 	id = "lava-base"
 	description = "A secret base researching illegal bioweapons, it is closely guarded by an elite team of syndicate agents."
 	suffix = "lavaland_surface_syndicate_base1.dmm"
 	cost = 20
 	allow_duplicates = FALSE
-
+*/
```
Comment (Azarak) [inline code/datums/ruins/lavaland.dm:64]:
```suggestion
*/
//SKYRAT EDIT REMOVAL END
```

File: code/game/area/areas/ruins/lavaland.dm [hunk 3]
```diff
@@ -81,6 +81,7 @@
 
 /area/ruin/unpowered/syndicate_lava_base/testlab
 	name = "Syndicate Lavaland Experimentation Lab"
+	area_flags = XENOBIOLOGY_COMPATIBLE //Skyrat Addition, Mapping
```
Comment (Azarak) [inline code/game/area/areas/ruins/lavaland.dm]:
```suggestion
	area_flags = XENOBIOLOGY_COMPATIBLE //SKYRAT EDIT ADDITION - MAPPING
```

**Comment by unit0016** [comment]
Noted.

**Review by Azarak — APPROVED** [review]

**Comment by Azarak** [comment]
@KathrinBailey

**Comment by unit0016** [comment]
Monkies got refactored, e. Will merge down to fix in a minute.

**Comment by unit0016** [comment]
Seems /mob/living/carbon/monkey is still valid, nevermind.
DOUBLE Nevermind, I opened strongdmm too early. Whoopsies. Fix incoming.

**Comment by Arctaisia** [comment]
Does it have turrets around it, too? Just to ask, because that was something some people got annoyed about with the current one, as that didn't have turrets.

**Comment by unit0016** [comment]
Yeah, it has the turrets still.

**Comment by Gandalf2k15** [comment]
I think @KathrinBailey can review and merge this.

**Pull request merged**